### PR TITLE
`Development`: Add documentation for the Cypress Bamboo build plans

### PR DESCRIPTION
**Pull request by jhainzlschmid**
### Description

In this PR I added documentation for the Bamboo buildplans executing the Cypress test suite.

### Steps for testing

1. [Build the Artemis documentation locally](https://github.com/ls1intum/Artemis/blob/documentation/cypress-deployment/docs/README.md)
2. Open the documentation html. You should see a new section in the Contributor Guide for the cypress bamboo build plans.
3. Check the cypress build plan documentation for typos and if it is understandable

#### Documentation Review
- [x] Review 1
- [x] Review 2

#### Screenshots
The deployment diagram for the Cypress Github Integration
cypress_bamboo_deployment_diagram

The deployment diagram for the Artemis deployment on test server 3
cypress_test_environment_deployment_diagram

### CODE REVIEW

**Review by willergeorg — APPROVED** [review]
The additions look good to me and are understandable.

**Review by b-fein — COMMENTED** [review]
The explanations seem reasonable, I think other people more familiar with your data centre setup will be a better judge for that.

Therefore, I focused mostly on typos and formatting. There are a lot of inline comments, but most of them are basically the same in multiple places about the capitalisation of proper nouns.

File: docs/dev/cypress.rst [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+**Bamboo webhook**
+
+The Artemis instance deployed on the build agent is not publicly available to improve the security of this setup.
+However, in order to get the build result for programming exercise submission Artemis relies on a webhook from Bamboo to send POST requests to Artemis.
+To allow this an extra rule has been added to the firewall allowing only the Bamboo instance in the prelive system to connect to the Artemis instance in the build agent.
+
+**Timing**
+
+As mentioned above we want the Cypress test suite to be executed whenever new commits are pushed to a git branch. This has been achieved by adding the `Cypress Github build plan <https://bamboo.ase.in.tum.de/browse/ARTEMIS-AETG>`__ as a `child dependency <https://confluence.atlassian.com/bamboo/setting-up-plan-build-dependencies-289276887.html>`__ to the `Artemis Build build plan <https://bamboo.ase.in.tum.de/browse/ARTEMIS-WEBAPP>`__.
+The Artemis Build build plan is triggered whenever a new commit has been pushed to a branch.
```
[the remark this answers is not in the document]
Comment (jhainzlschmid) [reply docs/dev/cypress.rst:58]: Good idea

**Comment by jhainzlschmid** [comment]
> The explanations seem reasonable, I think other people more familiar with your data centre setup will be a better judge for that.
> 
> Therefore, I focused mostly on typos and formatting. There are a lot of inline comments, but most of them are basically the same in multiple places about the capitalisation of proper nouns.

You are right, thanks for all the suggestions 👍

**Review by willergeorg — DISMISSED** [review]

File: docs/dev/cypress.rst [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+   Therefore, the Artemis Cypress Docker container is configured to install all dependencies (using :code:`npm ci`) upon start. This will also install Cypress itself. Afterwards the Artemis Cypress test suite is executed.
+   
+   The necessary configuration for the Cypress test suite is also passed in via environment variables. Furthermore, the Cypress container depends on the Artemis container and is only started once Artemis has been fully booted.
+
+**Bamboo webhook**
+
+The Artemis instance deployed on the build agent is not publicly available to improve the security of this setup.
+However, in order to get the build results for programming exercise submissions Artemis relies on a webhook from Bamboo to send POST requests to Artemis.
+To allow this an extra rule has been added to the firewall allowing only the Bamboo instance in the prelive system to connect to the Artemis instance in the build agent.
```
Comment (coaxial1) [inline docs/dev/cypress.rst:54]:
```suggestion
To allow this, an extra rule has been added to the firewall allowing only the Bamboo instance in the prelive system to connect to the Artemis instance in the build agent.
```

File: docs/dev/cypress.rst [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+**Timing**
+
+As mentioned above we want the Cypress test suite to be executed whenever new commits are pushed to a Git branch. This has been achieved by adding the `Cypress Github build plan <https://bamboo.ase.in.tum.de/browse/ARTEMIS-AETG>`__ as a `child dependency <https://confluence.atlassian.com/bamboo/setting-up-plan-build-dependencies-289276887.html>`__ to the `Artemis Build build plan <https://bamboo.ase.in.tum.de/browse/ARTEMIS-WEBAPP>`__.
```
Comment (coaxial1) [inline docs/dev/cypress.rst:58]:
```suggestion
As mentioned above, we want the Cypress test suite to be executed whenever new commits are pushed to a Git branch. This has been achieved by adding the `Cypress Github build plan <https://bamboo.ase.in.tum.de/browse/ARTEMIS-AETG>`__ as a `child dependency <https://confluence.atlassian.com/bamboo/setting-up-plan-build-dependencies-289276887.html>`__ to the `Artemis Build build plan <https://bamboo.ase.in.tum.de/browse/ARTEMIS-WEBAPP>`__.
```

**Review by coaxial1 — COMMENTED** [review]
Only found 2 missing commas :)

**Review by coaxial1 — APPROVED** [review]
LGTM!

**Review by willergeorg — APPROVED** [review]

**Review by cremertim — APPROVED** [review]
docs lgmt.

**Review by Bofan-Zhu — APPROVED** [review]
Documention is well-written and easily understandable. Looks good to me 👍

**Review by MertAksehirlioglu — DISMISSED** [review]
Documentation looks good to me

**Review by akesfeden — DISMISSED** [review]
Docs are understandable and descriptive 👍

**Review by baranimo7 — APPROVED** [review]
looks good to me!

**Pull request merged**